sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -198,14 +257,46 @@ class StreamExecution(
     logDebug(s"Unblocked at $newOffset for $source")
   }
 
+  override def awaitTermination(): Unit = {
+    require(state != INITIALIZED,
```

Comment:
Technically I think `require` should be reserved for checking arguments since it throws and `IllegalArgumentException`